docs/fpc-cc-invocation.puml

```diff
@@ -0,0 +1,158 @@
+@startuml
+
+'- force wrapping of longer text
+'skinparam wrapWidth 400
+
+!pragma teoz true
+
+hide footbox
+title Foot Box removed
+
+title Fabric Private Chaincode - Chaincode invocation
+
+actor "User Org1" as User1 order 9
+participant "ECC_Manager (syscc) Peer1_Org1" as Manager1 order 20
+
+participant "TLCC_Enclave Peer1_Org1" as TLCC1 order 25 #99FF99
+
+participant "ECC Peer1_Org1" as ECC1 order 30
+participant "ECC_Enclave" as Enclave1 order 30 #99FF99
+
+participant "ERCC Peer1_Org1" as ERCC1 order 50
+
+'participant "ECC Peer1_Org2" as ECC2 order 60
+'participant "ERCC Peer1_Org2" as ERCC2 order 62
+'
+'participant "ECC Peer1_Org3" as ECC3 order 70
+'participant "ERCC Peer1_Org3" as ERCC3 order 72
+
+ref over User1, ERCC1
+  see fpc-registration-new.puml
+end ref
+|||
+
+activate User1
+
+User1 -> ERCC1 ++: getECCInfo(CC_NAME)
+note right User1: The user may retrieve the ECC info for a specific <CC_NAME, ...>
+return ECCInfo = List<<ECC1_S_PK, ECC1_E_PK, ... , ATTESTATION>, ...>
+
+User1 -> User1 : verify ECCInfo
+note right #FFaa00
+  Only necessary iff User1 is not member of Org1 (otherwise authenticated response enough)
+  In that case, in general user also has to query different peers from different orgs,
+  enough to satisfy a quorum as necessary for the lifecycle endorsement policy and channel
+  configuration changes.
+  In this case, though, assuming client has a reference ECC_MR, TLCC_MR, CH_id
+  it might be sufficient to perform verification done in ""verify and store"" by peers in
+  ""register attestation"" flow.
+  (TBD)
+end note
+
+
+loop for all/"enough" endorsers
+
+  User1 -> User1 : args_ECC1 <- encrypt args with ECC1_E_PK
+
+  User1 -> ECC1 ++: send proposal (args_ECC1)
+
+  group invoke
+
+    ECC1 -> Enclave1 ++: check enclave is running
+    return
+
+    alt enclave does not exists
+      ECC1 -> Manager1 ++: get sealed state
+      note right Manager1 : get from local storage
+      return
+      ECC1 -> Enclave1 ++ : start enclave with sealed state
+      Enclave1 -> Enclave1 : unseal state and restore
+      Enclave1 <-> TLCC1: establish attested secure channel
+      note over TLCC1
+        protocol is SIGMA with 1-way local attestation of TLCC
+        with reportbody = <CH_id, TLCC_inst_id> where
+        TLCC_inst_id as unique (randomly generated) identifier of
+        this particular TLCC __instance__
+      end note
+      note over Enclave1
+        Verifies local attestation against TLCC_MR and CH_id
+        from sealed state. Remembers TLCC_inst_id.
+      end note
+      return
+    else
+      |||
+
+      ECC1 -> Enclave1 ++: invoke (args_ECC)
+
+      note right
+        Eventually we probably will also first verify signature of proposal?
```

Comment:
yes, if we have to authenticate clients based on MSP, and in this case we can rely on tlcc, which should already have (1) the data/certificates, (2) the MSP logic to validate it.